text/0000-data-formats.md

```diff
@@ -0,0 +1,151 @@
+
+## Requirements
+
+
+### Handle complex data
+
+Semantic information is often complex and multi-valued. How we represent semantic data is related to how attributes are defined at the logical layer.
+
+*   At minimum, nested lists of key/value pairs are necessary to represent them.
+*   Is it advantageous to also have strongly typed data structures?
+*   Are there issues with having strongly typed data structures in certain environment, such as dynamic languages?
+
+
+### Compose Multiple Semantics
+
+Objects in OpenTelemetry may describe more than one type of operation. For example, a span may simultaneously represent a MySQL transaction and a TCP/IP request. Pieces of metadata must compose well, without losing the structure integrity of each piece or risking a conflict in meaning.
+
+
+### Easily Extensible
+
+Not all semantics conventions and common operations are known at this time. It must be possible to add more semantic definitions without creating to versioning or compatibility issues with OpenTelemetry implementations.
+
+
+## Examples of Common Operations
+
+#### RPC calls
+
+Modeling distributed transactions is a core usecase, it is common to analyze  errors and latency for RPC calls. Because multiple individual network requests may be involved in a single logical RPC operation, it must be possible to describe RPC calls as logical operations, independent from the descriptions of transport and application protocols which they contain.
+
+Examples:
+*   A client sending a request to a remote server.
+*   A server receiving a request from a remote client.
+*   Logical vs physical networking: a request for “[https://www.wikipedia.org/wiki/Rabbit](https://www.wikipedia.org/wiki/Rabbit)” results in two physical HTTP requests - one 301 redirection, and another 200 okay. The  logical RPC operation represents both requests together.
+
+#### Network protocols
+
+The internet suite of network protocols is thoroughly standardized, and the details of network interactions calls are critical to analysing the behavior of distributed systems.
+
+Examples:
+*   Application Protocols: HTTP/1.1, HTTP/2, gRPC
+*   Transport Protocols: TCP, UDP
+*   Internet Protocols: IPv4, IPv6
+
+#### Database protocols
+
+Database requests often contain information which is key to root causing errors and latency.
+
+Examples:
+*   SQL: MySQL, Postrgres
+*   NoSQL: MongoDB
+
+
+# Logical Structure
+
+OpenTelemetry observability model, defined as data structures instead of API calls. A logical layer describes the default data structures expected to be produced by OpenTelemetry APIs for transport.
+
+## Requirements
+
+### Support Metadata
+
+The logical structure must be able to support all of the requirements for the metadata layer.
```

Comment:
again, can we avoid using "metadata"?

Reply:
I think we could just call the top layer "Semantic Conventions" but this layer needs to mention "metadata support" as that is the facility which the logical structure must have, in order for semantic conventions to be implemented.

Is it the concept you don't like, or just the term "metadata"?

Reply:
Dapper-style tracing systems are often referred to as "based on metadata propagation", so I find the usage of "metadata" in the context of trace data confusing, so best if we can avoid id. It's also not like there is some agreed upon understanding of what "metadata" means. Since the RFC talks about layered approach, how about
  * semantic layer
  * logical layer
  * ??? layer
  * transport layer

Reply:
How about "schema layer"...